### ISSUE TITLE
Decide what characters are allowed in station and device names.

### ISSUE THREAD

**Issue by HaroldMills** [HaroldMills/Vesper#149]
Vesper currently does not support + characters in station names, and perhaps other characters as well. For example, for the clip calendar URL for a station named "Station+", Vesper attempts to show a calendar for the station "Station" instead. No explicit decision has ever been made about exactly what characters are allowed in station and device names. We should make that decision, document it, and ensure that Vesper supports the chosen character set.

**Comment by HaroldMills** [comment]
A user recently reported a problem with a station name that included an apostrophe. In particular, I have verified that a clip calendar displays properly for such a station but a clip album invoked by clicking on a day in the calendar does not.

**Comment by HaroldMills** [comment]
This issue was fixed in c9d10b9707b25b52d47320a4bdb00b21f63a42f5. I decided that for now, at least, Vesper should endeavor to support all printable, non-whitespace ASCII characters and spaces in station, device, processor, and annotation names, and in annotation values. As far as I know, the referenced commit accomplishes that. It might be a good idea to try to expand this from ASCII to, say, UTF-8 in the future. Care will be required to ensure that this support is maintained, especially when encoding URLs in the client. See the JavaScript `ClipAlbum._onFilterClipsModalOkButtonClick` method for an example of proper URL encoding.
